gateway/mw_jwt.go

```diff
@@ -547,6 +568,7 @@ func (k *JWTMiddleware) processCentralisedJWT(r *http.Request, token *jwt.Token)
 		}
 	}
 
+	session.OauthClientID = oauthClientID
```

Comment:
There is really easy final step we can do:
```
if client, err := k.Spec.OAuthManager.OsinServer.Storage.GetClient(session.OauthClientID); err == nil {
  if developer_id, found := client.MetaData['developer_id']; found {
    session.DeveloperId = developer_id
  }
}
```

Reply:
I've been able to get this working and I'm guessing the goal is to make it available inside the session metadata object.
I don't think this change would benefit the analytics side for now ([PR](https://github.com/TykTechnologies/tyk-analytics/pull/2245)) as we don't use this value on the analytics record itself, only OAuth client ID is used.

Reply:
Solved